### PR TITLE
Update to 2024 beta

### PR DESCRIPTION
**Pull request by outsidermm**
Update requirement.txt and Pipfile to use the 2024 version of imported modules. (Not phoenix 6)

### CODE REVIEW

**Comment by outsidermm** [comment]
I'm not sure what to do with the CTRE current version number. I thought the 2024 is now called phoenix6, and we are using that later. If we use the 2023.1.0 CTRE it would have conflict with other dependencies as all the other dependencies it requires are 2023. (obviously)

**Comment by auscompgeek** [comment]
> I'm not sure what to do with the CTRE current version number. I thought the 2024 is now called phoenix6

We need to update to the 2024 beta version of Phoenix 5 first.

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,6 @@
-numpy~=1.24.1
-robotpy[apriltag,cscore]~=2023.4.2
-robotpy-ctre~=2023.1.0
-robotpy-navx~=2023.0.3
-robotpy-rev~=2023.1.3.2
-robotpy-photonvision~=2023.3.0
-robotpy-wpilib-utilities~=2023.1.0
+numpy~=1.26.0
+robotpy[apriltag,cscore]~=2024.0.0b3
```
Comment (auscompgeek) [inline requirements.txt:2]: Please update to the latest beta (currently `2024.0.0b4`).

File: requirements.txt [hunk 2]
```diff
@@ -1,7 +1,6 @@
...
+robotpy-ctre~=2024.0.0b1
```
Comment (auscompgeek) [inline requirements.txt:3]: We should use the latest beta here. There aren't any guarantees older vendor beta libraries will work with newer roboRIO/WPILib betas.

**Review by auscompgeek — CHANGES_REQUESTED** [review]

File: components/chassis.py [hunk 3]
```diff
@@ -19,7 +19,7 @@
 from magicbot import feedback
 
 from utilities.functions import constrain_angle, rate_limit_module
-from utilities.ctre import FALCON_CPR, FALCON_FREE_RPS
+from utilities.phoenix5 import FALCON_CPR, FALCON_FREE_RPS
```
Comment (auscompgeek) [inline components/chassis.py:22]: Let's not rename our `utilities.ctre` module.

**Comment by outsidermm** [comment]
Robot moves in sim

**Review by james-ward — APPROVED** [review]

**Review by auscompgeek — APPROVED** [review]

**Pull request merged**